pandas/core/internals.py

```diff
@@ -1697,24 +1698,20 @@ class NonConsolidatableMixIn(object):
     _holder = None
 
     def __init__(self, values, placement, ndim=None, fastpath=False, **kwargs):
+        # Placement must be converted to BlockPlacement so that we can check
+        # its length
+        if not isinstance(placement, BlockPlacement):
+            placement = BlockPlacement(placement)
```

Comment:
I found it much clearer to just copy these two lines rather than calling `mgr_locs.setter`, but can revert to that if desired.